c++/src/capnp/membrane.c++

```diff
@@ -111,62 +111,62 @@ public:
 
 private:
   _::CapTableBuilder* inner = nullptr;
-  MembranePolicy& policy;
+  kj::Rc<MembranePolicy> policy;
   bool reverse;
 };
 
 class MembranePipelineHook final: public PipelineHook, public kj::Refcounted {
 public:
   MembranePipelineHook(
-      kj::Own<PipelineHook>&& inner, kj::Own<MembranePolicy>&& policy, bool reverse)
+      kj::Own<PipelineHook>&& inner, kj::Rc<MembranePolicy> policy, bool reverse)
       : inner(kj::mv(inner)), policy(kj::mv(policy)), reverse(reverse) {}
 
   kj::Own<PipelineHook> addRef() override {
     return kj::addRef(*this);
   }
 
   kj::Own<ClientHook> getPipelinedCap(kj::ArrayPtr<const PipelineOp> ops) override {
-    return membrane(inner->getPipelinedCap(ops), *policy, reverse);
+    return membrane(inner->getPipelinedCap(ops), policy.addRef(), reverse);
   }
 
   kj::Own<ClientHook> getPipelinedCap(kj::Array<PipelineOp>&& ops) override {
-    return membrane(inner->getPipelinedCap(kj::mv(ops)), *policy, reverse);
+    return membrane(inner->getPipelinedCap(kj::mv(ops)), policy.addRef(), reverse);
   }
 
 private:
   kj::Own<PipelineHook> inner;
-  kj::Own<MembranePolicy> policy;
+  kj::Rc<MembranePolicy> policy;
   bool reverse;
 };
 
 class MembraneResponseHook final: public ResponseHook {
 public:
   MembraneResponseHook(
-      kj::Own<ResponseHook>&& inner, kj::Own<MembranePolicy>&& policy, bool reverse)
-      : inner(kj::mv(inner)), policy(kj::mv(policy)), capTable(*this->policy, reverse) {}
+      kj::Own<ResponseHook>&& inner, kj::Rc<MembranePolicy> policy, bool reverse)
+      : inner(kj::mv(inner)), policy(kj::mv(policy)), capTable(this->policy.addRef(), reverse) {}
 
   AnyPointer::Reader imbue(AnyPointer::Reader reader) { return capTable.imbue(reader); }
 
 private:
   kj::Own<ResponseHook> inner;
-  kj::Own<MembranePolicy> policy;
+  kj::Rc<MembranePolicy> policy;
```

Comment:
This member is no longer needed, since you're passing a strong ref directly into `capTable`, and this member isn't otherwise used.